Add option to add/remove a specific port/env var of a specific container

valaparthvi opened the pull request:
### What does this PR do?:
<!-- _Summarize the changes_ -->
This PR modifies devfileObj methods such as SetPorts to allow setting port to a specific container, and RemovePorts to remove a specific port of a specific container.
### Which issue(s) this PR fixes:
<!-- _Link to github issue(s)_ -->
https://github.com/redhat-developer/odo/issues/5423
### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed


- [x] Unit/Functional tests

  <!-- _These are run as part of the PR workflow, ensure they are updated_ -->

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  <!--  _Do we need to verify integration with ODO and Openshift console?_ -->

- [ ] Documentation 

   <!-- _This includes product docs and READMEs._ -->

- [ ] Client Impact

  <!-- _Do we have anything that can break our clients?  If so, open a notifying issue_ -->


### How to test changes / Special notes to the reviewer:
The unit tests should help give an idea about the PR changes.

yangcao77 commented on pkg/devfile/parser/configurables.go:
```diff
@@ -60,17 +60,19 @@ func (d DevfileObj) RemoveEnvVars(keys []string) (err error) {
-// SetPorts converts ports to endpoints, adds to a devfile
-func (d DevfileObj) SetPorts(ports ...string) error {
+// SetPorts accepts a map of container name and the port numbers to be set;
+// it converts ports to endpoints, and adds to a devfile
```
please update the description as this PR changes the existing behavior. currently it will set the `ports` to all container components, and this change will look for specific component and add the endpoint.

valaparthvi commented:
/hold

valaparthvi commented:
/unhold

valaparthvi commented:
@yangcao77 I have modified the existing devfileObj methods and added a few new methods to the `DevfileData` interface. Reason being, we wanted methods that would update the devfile data without writing to the actual file.

yangcao77 commented on pkg/devfile/parser/data/v2/containers.go:
```diff
@@ -0,0 +1,265 @@
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (d *DevfileV2) AddEnvVars(containerEnvMap map[string][]v1alpha2.EnvVar) error {
```
Can you add some comments for those exposed functions?

yangcao77 reviewed (approved):
Changes look good to me.
